packages/protocol/test/governance/vesting.ts

```diff
@@ -0,0 +1,1734 @@
+            vestingInstance.setAccountMetadataURL(metadataURL, { from: beneficiary })
+          )
+        })
+
+        it('should revert if non-revoker attempts to set the metadataURL', async () => {
+          await assertRevert(
+            vestingInstance.setAccountMetadataURL(metadataURL, { from: accounts[6] })
+          )
+        })
+      })
+    })
+  })
+
+  describe('#setAccountDataEncryptionKey()', () => {
+    let vestingInstanceRegistryAddress
+    let vestingInstance
+    const dataEncryptionKey = '0x02f2f48ee19680706196e2e339e5da3491186e0c4c5030670656b0e01611111111'
+    const longDataEncryptionKey =
+      '0x04f2f48ee19680706196e2e339e5da3491186e0c4c5030670656b0e01611111111' +
+      '02f2f48ee19680706196e2e339e5da3491186e0c4c5030670656b0e01611111111'
+
+    beforeEach(async () => {
+      await createNewVestingInstanceTx(vestingDefaultSchedule, web3)
+      vestingInstanceRegistryAddress = await vestingFactoryInstance.vestings(beneficiary)
+      vestingInstance = await VestingInstance.at(vestingInstanceRegistryAddress)
+      await vestingInstance.createAccount({ from: beneficiary })
+    })
+
+    it('beneficiary should set dataEncryptionKey', async () => {
+      // @ts-ignore
+      await vestingInstance.setAccountDataEncryptionKey(dataEncryptionKey, { from: beneficiary })
+      // @ts-ignore
+      const fetchedKey: string = await accountsInstance.getDataEncryptionKey(
+        vestingInstance.address
+      )
+      assert.equal(fetchedKey, dataEncryptionKey)
+    })
+
+    it('should revert if non-beneficiary attempts to set dataEncryptionKey', async () => {
+      // @ts-ignore
+      await assertRevert(
+        vestingInstance.setAccountDataEncryptionKey(dataEncryptionKey, { from: accounts[2] })
+      )
+    })
+
+    it('should allow setting a key with leading zeros', async () => {
+      const keyWithZeros = '0x00000000000000000000000000000000000000000000000f2f48ee19680706191111'
+      // @ts-ignore
+      await vestingInstance.setAccountDataEncryptionKey(keyWithZeros, { from: beneficiary })
+      // @ts-ignore
+      const fetchedKey: string = await accountsInstance.getDataEncryptionKey(
+        vestingInstance.address
+      )
+      assert.equal(fetchedKey, keyWithZeros)
+    })
+
+    it('should revert when the key is invalid', async () => {
+      // @ts-ignore
+      await assertRevert(
+        vestingInstance.setAccountDataEncryptionKey('0x32132931293', { from: beneficiary })
+      )
+    })
+
+    it('should allow a key that is longer than 33 bytes', async () => {
+      // @ts-ignore
+      await vestingInstance.setAccountDataEncryptionKey(longDataEncryptionKey, {
+        from: beneficiary,
+      })
+      // @ts-ignore
+      const fetchedKey: string = await accountsInstance.getDataEncryptionKey(
+        vestingInstance.address
+      )
+      assert.equal(fetchedKey, longDataEncryptionKey)
+    })
+  })
+
+  Object.keys(authorizationTestDescriptions).forEach((key) => {
```

Comment:
nit: These are very nice but ultimately are a bit more detailed than is warranted, IMO. If they become a pain to maintain, feel free to remove in favor of a more simple test that checks that authorizations can be made

Reply:
ok, thanks, noted, this structure allows further tests to be added if necessary.